realm-transformer/src/main/groovy/io/realm/transformer/BytecodeModifier.groovy

```diff
@@ -152,11 +152,12 @@ class BytecodeModifier {
 
     private static CtMethod createSetter(String methodName, CtField field) {
         CtMethod setter = CtNewMethod.setter(methodName, field)
-//        if ("io.realm.RealmInteger".equals(field.getType().getClass().getName())) {
-//            getter.insertBefore(
-//                    'if (!($1 instanceof io.realm.internal.counters.UnmanagedRealmInteger))' +
-//                    'throw new IllegalStateException("Cannot assign a managed RealmInteger to an unmanaged object")');
-//        }
+        logger.debug "##### SETTER: for ${methodName}: " + field.getType().getName()
+        if ("io.realm.RealmInteger".equals(field.getType().getName())) {
```

Comment:
I thought so too!   I think, though, that since this happens at compile time, way before Proguard gets its evil little hands on things, it will work fine.